e2e_playwright/st_pydeck_chart_select_test.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+from e2e_playwright.shared.app_utils import click_button
+
+
+# Firefox seems to be failing but can't reproduce locally and video produces an empty page for firefox
+@pytest.mark.skip_browser("firefox")
```

Comment:
Quite interesting. When you say "Can't reproduce locally" is that using playwright locally with the firefox browser?

Reply:
Full transparency, I copied this from the existing PyDeck tests: https://github.com/streamlit/streamlit/blob/develop/e2e_playwright/st_pydeck_chart_test.py#L22

Reply:
After spending a bunch of time debugging different test failures in CI vs local, I've decided to only test these in CI in Chromium. Firefox fails on CI but I can't reproduce locally and video from CI produces an empty element for PyDeck. Webkit is too slow on CI, although everything passes locally.

Getting coverage in Chromium is sufficient for now since the underlying logic is the same and we are not testing for browser-specific behavior, and all of that is ultimately handed off to deck.gl anyway.

Additionally, even the deck.gl docs caution against visual tests since it renders to a canvas element: https://github.com/visgl/deck.gl/blob/master/docs/developer-guide/testing.md

I left notes about this in all of the test files for this.